_drafts/circle-two-point-oh.md

```diff
@@ -0,0 +1,93 @@
+---
+layout: post
+title: "Upgrading Volt to CircleCI 2.0"
+date: 2017-12-20
+comments: true
+author: jonallured
+categories: [ci]
+---
+I was really excited about CircleCI 2.0, especially the [workflows
+features][workflows]. It seemed to me that with the work they had done here,
+really complicated builds would be able to be configured in a way that made more
+sense than on 1.0. This was something that was causing me grief on one of our
+projects so I upgraded to 2.0. It was pretty hard to get green and once we did,
+we decided to downgrade back to 1.0. Here's why.
+
+[workflows]: https://circleci.com/docs/2.0/workflows/
+
+Volt is the internal name for the app that our Partners use to enter information
+about their artworks and artists into the Artsy platform. It's a Rails app and
+has a pretty big and slow test suite. Over time, we've ended up with a good
+number of things that run during CI for this project:
+
+* [Danger][]
+* [RuboCop][]
+* [Prettier][]
+* [Jest tests][Jest]
+* [Teaspoon tests][Teaspoon]
+* [RSpec tests][RSpec]
+
+[Danger]: http://danger.systems/
+[RuboCop]: http://rubocop.readthedocs.io/
+[Prettier]: https://prettier.io/
+[Jest]: https://facebook.github.io/jest/
+[Teaspoon]: https://github.com/jejacks0n/teaspoon
+[RSpec]: http://rspec.info/
+
+Some of these things are fast and some are slow. Ok, fine, really only the view
+specs are slow - running them in isolation takes around 20 minutes on my laptop.
+
+My goal was to separate out the fast stuff, run that in parallel and only if all
+that passed would I run the view specs. This should get feedback to developers
+faster when they've broken something and not waste time running the slowest part
+of the CI stack.
+
+Artsy has 7 containers with CircleCI and Volt takes up 6 of them, so being able
+to break apart a run into pieces seemed like a good thing for the wider Artsy
+engineering organization too - developers working on something else would spend
+less time blocked by Volt runs and developers working on Volt shouldn't notice
+anything. Foreshadowing!!
+
+Here's the CircleCI 2.0 [config][] I ended up with and here's a pretty picture
+of what our CircleCI 2.0 builds looked like:
+
+[config]: https://gist.github.com/jonallured/45032779506138186973af7cb94b5363
+
+![/images/circle-two-oh/volt-circle-two-workflow.png](/images/circle-two-oh/volt-circle-two-workflow.png)
```

Comment:
This uses the file name as an image description, want to update it for our visually-impaired readers?

Reply:
Sorry, I'm not sure I understand, can you point me towards a post that does this well?

Reply:
Well, I looked for other posts and couldn't find any _good_ examples. While both https://github.com/artsy/artsy.github.io/blob/source/_posts/2014-11-13-eidolon-retrospective.markdown and https://github.com/artsy/artsy.github.io/blob/source/_posts/2016-10-26-jwt-artsy-journey.md have image descriptions, but they don't really _describe_ the images. It looks like we could do better on this on the Artsy blog in general, and I didn't mean to single you out here Jon.

I would replace this line with something like:

```markdown
![A screenshot of Circle CI 2.0, showing parallel tasks](/images/circle-two-oh/volt-circle-two-workflow.png)
```

Reply:
Ohhh, I see - I guess my Markdown syntax knowledge fell down here. Will update!